02/app.js

```diff
@@ -1,19 +1,33 @@
-document.addEventListener('DOMContentLoaded', init);
+document.addEventListener("DOMContentLoaded", init);
 
 function init() {
-    const divList = document.querySelectorAll('div');
-    
-    setBorderColorAsync(divList[0], 'red', function() {
-        setBorderColorAsync(divList[1], 'blue', function() {
-            setBorderColorAsync(divList[2], 'green', function() {
-                console.log('finish');
-            });
-        });
-    });
+  const divList = document.querySelectorAll("div");
 
+  setBorderColorAsync(divList[0], "red", () => {})
+    .then(() => setBorderColorAsync(divList[1], "blue", function () {}))
+    .then(() => setBorderColorAsync(divList[2], "green", function () {}))
+    .catch((err) => console.error(err))
+    .finally(() => console.log("finish"));
 }
 
 function setBorderColorAsync(element, color, callback) {
+  if (element && element instanceof HTMLElement) {
```

Comment:
Raczej ten `if` powinien być wew. obietnicy. Wtedy jeśli nie jest ten warunek sprawdzony to uruchamiamy `reject()`